internal/resolution/pm/gradle/job.go

```diff
@@ -77,3 +88,122 @@ func (j *Job) Run() {
 func (j *Job) GetDir() string {
 	return j.dir
 }
+
+func (j *Job) handleError(cmdErr job.IError) {
+	expressions := []string{
+		bugErrRegex,
+		notRootDirErrRegex,
+		unrelatedBuildErrRegex,
+		unknownPropertyErrRegex,
+	}
+
+	for _, expression := range expressions {
+		regex := regexp.MustCompile(expression)
+
+		if regex.MatchString(cmdErr.Error()) {
+			cmdErr = j.addDocumentation(expression, regex, cmdErr)
+			j.Errors().Append(cmdErr)
+
+			return
+		}
+	}
+
+	j.Errors().Append(cmdErr)
+}
+
+func (j *Job) addDocumentation(expr string, regex *regexp.Regexp, cmdErr job.IError) job.IError {
+	switch {
+	case expr == bugErrRegex:
+		cmdErr = j.addBugErrorDocumentation(regex, cmdErr)
+	case expr == notRootDirErrRegex:
+		cmdErr = j.addNotRootDirErrorDocumentation(regex, cmdErr)
+	case expr == unrelatedBuildErrRegex:
+		cmdErr = j.addUnrelatedBuildErrorDocumentation(regex, cmdErr)
+	case expr == unknownPropertyErrRegex:
+		cmdErr = j.addUnknownPropertyErrorDocumentation(regex, cmdErr)
+	}
+
+	return cmdErr
```

Comment:
Opinion: I think the called functions here could return the documentation string instead so we only need one call to SetDocumentation. But I'm not sure what is best!